pkg/products/monitoring/prometheusRules.go

```diff
@@ -0,0 +1,287 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *Reconciler) reconcileBackupMonitoringAlerts(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
+	monitoringConfig := config.NewMonitoring(config.ProductConfig{})
+	rule := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "backup-monitoring-alerts",
+			Namespace: r.Config.GetOperatorNamespace(),
+		},
+	}
+
+	rules := []monitoringv1.Rule{
+		{
+			Alert: fmt.Sprintf("JobRunningTimeExceeded"),
+			Annotations: map[string]string{
+				"sop_url": "https://github.com/RHCloudServices/integreatly-help/blob/master/sops/alerts_and_troubleshooting.md",
+				"message": " Job {{ $labels.namespace }} / {{ $labels.job  }} has been running for longer than 300 seconds",
+			},
+			Expr:   intstr.FromString("time() - (max(kube_job_status_active * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key='middleware'}) BY (job_name) * ON(job_name) GROUP_RIGHT() max(kube_job_status_start_time * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key='middleware'}) BY (job_name, namespace, label_cronjob_name) > 0) > 300 "),
```

Comment:
The double quotes were removed because it caused errors in the code version of the alerts so thats why single quotes were used. I have no idea how double single quotes are being created all the single quotes in the expressions are single "single" quotes

Reply:
Yeah, I don't know what's going on there either...the fact that it's not happening for the other PrometheusRule CRs makes it stranger.

I think maybe it's because the entire query is wrapped in single quotes (no idea why, you don't have it like that in the code):

```yaml
expr: 'time() - (max(kube_job_status_active * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key=''middleware''})
        BY (job_name) * ON(job_name) GROUP_RIGHT() max(kube_job_status_start_time
        * ON(job_name) GROUP_RIGHT() kube_job_labels{label_monitoring_key=''middleware''})
        BY (job_name, namespace, label_cronjob_name) > 0) > 300 '
```

Given that once the alerts appear in Prometheus fine (normal double quoting), I think we can ignore this.